hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -534,6 +535,15 @@ object HoodieFileIndex extends Logging {
       properties.setProperty(RECORDKEY_FIELD.key, tableConfig.getRecordKeyFields.orElse(Array.empty).mkString(","))
       properties.setProperty(PRECOMBINE_FIELD.key, Option(tableConfig.getPreCombineField).getOrElse(""))
       properties.setProperty(PARTITIONPATH_FIELD.key, HoodieTableConfig.getPartitionFieldPropForKeyGenerator(tableConfig).orElse(""))
+
+      // for simple bucket index, we need to set the INDEX_TYPE, BUCKET_INDEX_HASH_FIELD, BUCKET_INDEX_NUM_BUCKETS
+      val dataBase = Some(tableConfig.getDatabaseName)
```

Comment:
all of these properties are write configs, so we wanna fix it by explicitly setting up all the catalog properties.

Reply:
Yes, that's what I think. I can't think of any other ways to get these properties.

Reply:
@danny0405 @xicm hello, We consider a scenario where when creating a table, the number of buckets is not specified, and when writing data in Flink mode, SQL Hint is used to specify the number of buckets. Then, do we have to specify the number of buckets when reading Spark? If not specified, when optimizing the Spark Bucket Index read, an incorrect number of buckets will be obtained, resulting in abnormal data queries.

Reply:
> SQL Hint is used to specify the number of buckets. Then, do we have to specify the number of buckets when reading Spark

You should because bucket number is a write config now. And it can not be changed once it is setup.

Reply:
@danny0405 hello, Is it not user-friendly to require the user to display a specified number of buckets when reading. Can we save the bucket count when writing data, such as writing it to table config, in the scenario where the bucket count is specified in SQL Hint but not specified during table creation?
In this way, downstream users do not need to display the specified bucket count when reading the hudi table, as they may not be concerned about the bucket count of this table.